app/services/v1/sms/replay.rb

```diff
@@ -33,4 +46,71 @@ def delete_messaged_for(number)
       job.delete if job_args['job_class'] == 'SmsPlans::SendSmsJob' && job_args['arguments'].first.eql?(number)
     end
   end
+
+  def handle_message_with_sms_code
+    session = Session::Sms.includes(:intervention).where(intervention: { status: :published }).find_by(sms_code: message)
+    return SmsPlans::SendSmsJob.perform_later(from_number, I18n.t('sms.session_not_found'), nil, @user&.id) unless session
+
+    if @user
+      user_session = UserSession::Sms.find_by(session_id: session.id, user_id: @user.id)
```

Comment:
We can allow filling the same session multiple times. In that case, find_by will be insufficient.

<img width="804" alt="Screenshot 2024-05-20 at 19 11 25" src="https://github.com/Michigan-State-University/cias-api/assets/61981027/cf9efeeb-0c62-498e-8c30-a657334d2cdf">